v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/avro/GenericRecordTypeConvertor.java

```diff
@@ -136,10 +167,30 @@ public Map<String, Value> transformChangeEvent(GenericRecord record, String srcT
             e);
       }
     }
-    result = populateCustomTransformations(result, record, srcTableName);
     return result;
   }
 
+  private String getUUID() {
+    return UUID.randomUUID().toString();
+  }
+
+  // Returns true if the column exists at source. For identity mapper, it checks if the column
+  // exists in the GenericRecord.
+  private boolean colExistsAtSource(
```

Comment:
this seems like a leaky abstraction. We should not have switch cases based on types of mappers here.

Can we add a method in the schema mapper like hasSourceColumn or so skip if not present ?

Reply:
That was my original intent but we cannot do that as the IdentityMapper is not aware of any source specific schema details. For this, we rely on the generic record to give us the source column names which is not ideal.

In the long term, once we develop capabilities to read the source schema in the job itself, we can use that in the identity mapper and refactor this.

Reply:
Added a comment in the code.

Reply:
I wouldn't completely agree on this. 

I think we should do both these checks for all mappers in any case.
Get the sourceColumn, in the identity mapper it would just return the same name.

And then you check if the record has the sourceColumn - regardless of the mapper.